### PR TITLE
SDL2: Add `-no-undefined` to LD_FLAGS for libSDL2main

### PR DESCRIPTION
**Pull request by NHDaly**
Add `-no-undefined` to LD_FLAGS for `libSDL2main` (and `libSDL2_test.la`)

Attempting to fix problems in https://github.com/JuliaPackaging/Yggdrasil/pull/123

### CODE REVIEW

File: S/SDL2/build_tarballs.jl [hunk 1]
```diff
@@ -18,6 +18,9 @@ cd $WORKSPACE/srcdir/SDL2-*/
 FLAGS=()
 if [[ "${target}" == *-linux-* ]] || [[ "${target}" == *-freebsd* ]]; then
     FLAGS+=(--with-x)
+elif [[ "${target}" == *-mingw* ]]; then
+    FLAGS+='libSDL2main_la_LDFLAGS=-no-undefined'
```
Comment (giordano) [inline S/SDL2/build_tarballs.jl]:
```suggestion
    FLAGS+=('libSDL2main_la_LDFLAGS=-no-undefined')
```

File: S/SDL2/build_tarballs.jl [hunk 2]
```diff
@@ -18,6 +18,9 @@ cd $WORKSPACE/srcdir/SDL2-*/
...
+    FLAGS+='libSDL2_test_la_LDFLAGS=-no-undefined'
```
Comment (giordano) [inline S/SDL2/build_tarballs.jl]:
```suggestion
    FLAGS+=('libSDL2_test_la_LDFLAGS=-no-undefined')
```

**Comment by NHDaly** [comment]
Huh, it's weird that half those builds failed, when i didn't change anything. Were those ever succeeding?

It's hard to tell, because it looks like CI wasn't set up when this package was first added?:
https://github.com/JuliaPackaging/Yggdrasil/pull/27


Also, it looks like the flag either wasn't picked up or wasn't effective, because the warning remains:
https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=780&view=logs&j=3b6dd8ee-51ff-5e19-5606-d30ecf5c3515&t=a88e4976-10a6-5bfe-92ae-249d2fa489a6&l=2353

**Comment by giordano** [comment]
We don't need this anymore, but we should figure out why those platforms are now failing....

**Pull request closed without merging**